core_impl/src/test/java/io/opencensus/trace/export/ActiveSpansExporterTest.java

```diff
@@ -0,0 +1,160 @@
+        .isEqualTo(1);
+    span1.end();
+    assertThat(activeSpansExporter.getSummary().getPerSpanNameSummary().size()).isEqualTo(1);
+    assertThat(activeSpansExporter.getSummary().getPerSpanNameSummary().get(SPAN_NAME_1)).isNull();
+    assertThat(
+            activeSpansExporter
+                .getSummary()
+                .getPerSpanNameSummary()
+                .get(SPAN_NAME_2)
+                .getNumActiveSpans())
+        .isEqualTo(1);
+    span2.end();
+    assertThat(activeSpansExporter.getSummary().getPerSpanNameSummary().size()).isEqualTo(0);
+  }
+
+  @Test
+  public void getSummary_SpansWithSameName() {
+    SpanImpl span1 = createSpan(SPAN_NAME_1);
+    SpanImpl span2 = createSpan(SPAN_NAME_1);
+    SpanImpl span3 = createSpan(SPAN_NAME_1);
+    assertThat(activeSpansExporter.getSummary().getPerSpanNameSummary().size()).isEqualTo(1);
+    assertThat(
+            activeSpansExporter
+                .getSummary()
+                .getPerSpanNameSummary()
+                .get(SPAN_NAME_1)
+                .getNumActiveSpans())
+        .isEqualTo(3);
+    span1.end();
+    assertThat(activeSpansExporter.getSummary().getPerSpanNameSummary().size()).isEqualTo(1);
+    assertThat(
+            activeSpansExporter
+                .getSummary()
+                .getPerSpanNameSummary()
+                .get(SPAN_NAME_1)
+                .getNumActiveSpans())
+        .isEqualTo(2);
+    span2.end();
+    assertThat(activeSpansExporter.getSummary().getPerSpanNameSummary().size()).isEqualTo(1);
+    assertThat(
+            activeSpansExporter
+                .getSummary()
+                .getPerSpanNameSummary()
+                .get(SPAN_NAME_1)
+                .getNumActiveSpans())
+        .isEqualTo(1);
+    span3.end();
+    assertThat(activeSpansExporter.getSummary().getPerSpanNameSummary().size()).isEqualTo(0);
+  }
+
+  @Test
+  public void getActiveSpans_SpansWithDifferentNames() {
+    SpanImpl span1 = createSpan(SPAN_NAME_1);
+    SpanImpl span2 = createSpan(SPAN_NAME_2);
+    assertThat(activeSpansExporter.getActiveSpans(Filter.create(SPAN_NAME_1, 0)))
+        .containsExactly(span1.toSpanData());
+    assertThat(activeSpansExporter.getActiveSpans(Filter.create(SPAN_NAME_1, 2)))
+        .containsExactly(span1.toSpanData());
+    assertThat(activeSpansExporter.getActiveSpans(Filter.create(SPAN_NAME_2, 0)))
+        .containsExactly(span2.toSpanData());
+    span1.end();
+    span2.end();
+  }
+
+  @Test
+  public void getActiveSpans_SpansWithSameName() {
+    SpanImpl span1 = createSpan(SPAN_NAME_1);
+    SpanImpl span2 = createSpan(SPAN_NAME_1);
+    SpanImpl span3 = createSpan(SPAN_NAME_1);
+    assertThat(activeSpansExporter.getActiveSpans(Filter.create(SPAN_NAME_1, 0)))
+        .containsExactly(span1.toSpanData(), span2.toSpanData(), span3.toSpanData());
+    assertThat(activeSpansExporter.getActiveSpans(Filter.create(SPAN_NAME_1, 2)))
+        .containsExactly(span2.toSpanData(), span3.toSpanData());
```

Comment:
Can users rely on getting only the latest spans when they set a maximum?  I think it would be better to only test that any two are returned, or document the behavior in the Filter class.